### ISSUE TITLE
Difference between build/generic/web 356Kb viewer.js and 169Kb pdfjs-dist/web/pdf_viewer.js

### ISSUE THREAD

**Issue by SwartHack** [mozilla/pdf.js#9318]
I believe I am missing something obvious, but have banged my head enough that I could really use some first-aid and help please. 

I have been using pdfjs-dist to create client-side thumbnails without problems. Very similar to the many excellent examples for doing this. I now built the generic build of pdf.js and want to start with that generic viewer , which runs great using the gulp server! I am using all the latest stuff...

Using the pdfjs-dist/web/pdf_viewer.js, along with the build/generic/web viewer.html I am able to load a document...

`function loadPdfFile(fileDataUrl) {

            var SEARCH_FOR = ''; 
            var mypdf = document.getElementById('my-pdf-viewer');
            var container = mypdf.querySelector('#viewerContainer');

            // (Optionally) enable hyperlinks within PDF files.
            var pdfLinkService = new PDFJS.PDFLinkService();
            
            var pdfViewer = new PDFJS.PDFViewer({
                container: container
            });
            pdfLinkService.setViewer(pdfViewer);

            // (Optionally) enable find controller.
            var pdfFindController = new PDFJS.PDFFindController({
                pdfViewer: pdfViewer
            });
            pdfViewer.setFindController(pdfFindController);

            container.addEventListener('pagesinit', function () {
                // We can use pdfViewer now, e.g. let's change default scale.
                pdfViewer.currentScaleValue = 'page-width';

                if (SEARCH_FOR) { // We can try search for things
                    pdfFindController.executeCommand('find', { query: SEARCH_FOR });
                }
            });

            // Loading document.
            PDFJS.getDocument(fileDataUrl).then(function (pdfDocument) {
                // Document loaded, specifying document for the viewer and
                // the (optional) linkService.
                pdfViewer.setDocument(pdfDocument);
                pdfLinkService.setDocument(pdfDocument, null);
            });
        }`
But all the controls are displaced and don't work.  Obviously as I am missing the required settings, hooks, etc. that are in the bigger file.

When I tried the bigger file, I get 

> .PDFLinkService() is not a constructor

After examining the similarities and differences of these files till dizzy, I just couldn't piece it out. Especially the different packaging?!  I also find myself wondering if I need to create new objects, as they should already be there in the viewer?! 

The big one instantiates everything on load?!, which is nice, but how do I hook into it?  I can't use a link with a query string as demonstrated in the examples as this is embedded asynchronously. 

What I want is the viewer to open without a default, just empty. Then I feed it the fileDataUrl with .getDocument.

If someone can set me straight on this, I'd really appreciate it. When I get it all worked out I'll post a link to a live example. I am also working with a SASS file I created and will share that when ready!

Cheers!

**Comment by timvandermeij** [comment]
The `pdf_viewer.js` (https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.js) file is, together with the `BaseViewer` class, responsible for the rendering of the pages in the default viewer at https://mozilla.github.io/pdf.js/web/viewer.html. The `viewer.js` file works with that and adds a lot of other bindings, such as all the buttons in the UI. The idea is that people can make their own viewer using the `BaseViewer` class so they don't need to have all the controls that the default viewer has. For instance, https://github.com/mozilla/pdf.js/blob/master/examples/components/singlepageviewer.js uses the `PDFSinglePageViewer`, which is a basic viewer component for rendering one page at a time. As you can see at https://github.com/mozilla/pdf.js/blob/master/web/pdf_single_page_viewer.js, it also extends the `BaseViewer` class.

In summary, the former is just a component to render the pages and the latter is a fully-fledged viewer application.

**Comment by TroyWolf** [comment]
The reply from @timvandermeij makes perfect sense to me except in what he describes, I would expect `viewer.js` to require `pdf_viewer.js`. Instead, the pre-built files don't include `pdf_viewer.js`.

So while I want to use viewer.js as a great example to develop my own, custom viewer, it is not clear how to create my own that incorporates pdf_viewer.js. I can't user viewer.js as an example because it does not _appear_ to use pdf_viewer.js.

Like @SwartHack, I feel I'm missing something fundamental in this mix. ☕️

**Comment by Bluejanis** [comment]
Also having both viewer.js and pdf_viewer.js is very confusing in a library that displays pdfs.
Renaming one of them would be huge improvement - and a simple one too 🦄
